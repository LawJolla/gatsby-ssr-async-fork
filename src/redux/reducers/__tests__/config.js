const reducer = require(`../config`)

describe(`config reducer`, () => {
  it(`let's you add a config`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {
        siteMetadata: {
          hi: true,
        },
      },
    }
    expect(reducer(undefined, action)).toMatchSnapshot()
  })

  it(`handles empty configs`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
    }
    expect(reducer(undefined, action)).toMatchSnapshot()
  })

  it(`Validates configs with unsupported options`, () => {
    const config = {
      someRandomThing: `hi people`,
      plugins: [],
    }
    function runReducer() {
      return reducer(
        {},
        {
          type: `SET_SITE_CONFIG`,
          payload: config,
        }
      )
    }
    expect(runReducer).toThrowErrorMatchingSnapshot()
  })

  it(`It corrects pathPrefixes without a forward slash at beginning`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {
        pathPrefix: `prefix`,
      },
    }
    expect(reducer(undefined, action).pathPrefix).toBe(`/prefix`)
  })

  it(`It removes trailing forward slash`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {
        pathPrefix: `/prefix/`,
      },
    }
    expect(reducer(undefined, action).pathPrefix).toBe(`/prefix`)
  })

  it(`It sets the pathPrefix to an empty string if it's not set`, () => {
    const action = {
      type: `SET_SITE_CONFIG`,
      payload: {},
    }
    expect(reducer(undefined, action).pathPrefix).toBe(``)
  })
})
