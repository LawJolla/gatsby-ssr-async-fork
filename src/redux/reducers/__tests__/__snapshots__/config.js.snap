// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config reducer Validates configs with unsupported options 1`] = `"The site's gatsby-config.js failed validation"`;

exports[`config reducer handles empty configs 1`] = `
Object {
  "pathPrefix": "",
  "polyfill": true,
}
`;

exports[`config reducer let's you add a config 1`] = `
Object {
  "pathPrefix": "",
  "polyfill": true,
  "siteMetadata": Object {
    "hi": true,
  },
}
`;
